### DISCUSSION TITLE
DuckDB does not recognize a `polars` `Enum` (or a `parquet` `ENUM`) as a DuckDB `ENUM`? Should I be creating an issue for this?

### DISCUSSION THREAD

**Discussion by bzm3r** [duckdb/duckdb#13674] [Q&A]
I was playing with the following example:

```py
import polars as pl

values = ["a", "b", "c"]
df = pl.DataFrame(pl.Series("x", values, dtype=pl.Enum(values)))
print(df)
```

```
shape: (3, 1)
┌──────┐
│ x    │
│ ---  │
│ enum │
╞══════╡
│ a    │
│ b    │
│ c    │
└──────┘
```

```py
import duckdb as ddb

ddb.sql("SELECT * FROM df").show()
```

```
┌─────────┐
│    x    │
│ varchar │
├─────────┤
│ a       │
│ b       │
│ c       │
└─────────┘
```

Why does DuckDB not convert `polars` `Enum` into a [DuckDB `Enum`](https://duckdb.org/docs/sql/data_types/enum.html)?

(Initially, I was trying to read from a `polars` dataframe I had saved as a `parquet` file on disk. On doing so, I ran into the same issue, and I wondered if the problem might have something to do with reading from `parquet` files, but the issue exists "reading directly" from `polars` too.)

Also: earching for "polars enum" or "parquet enum" does not seem to find any relevant issues, so I am not sure if I should be creating an issue here or not?

**Comment by cmdlineluser** [comment]
It seems Polars conversion is done via arrow.

Polars Categorical/Enum type is converted to an arrow dictionary type:

```python
>>> df.to_arrow()
pyarrow.Table
x: dictionary<values=large_string, indices=uint32, ordered=0>
----
x: [  -- dictionary:
["a","b","c"]  -- indices:
[0,1,2]]
```

There are several previously closed issues on that topic:

- https://github.com/duckdb/duckdb/issues/4268
- https://github.com/duckdb/duckdb/issues/4290

This means even if you go from DuckDB -> arrow -> DuckDB you will get a VARCHAR.

```python
q = duckdb.sql("select unnest(['a', 'b'])::enum('a', 'b') as col")

duckdb.from_arrow(q.to_arrow_table())
# ┌─────────┐
# │   col   │
# │ varchar │
# ├─────────┤
# │ a       │
# │ b       │
# └─────────┘
```

**Comment by soerenwolfers** [comment]
I'm aware it's not pretty and has other problems, but maybe converting to pandas first, using `.to_pandas()`, is good enough for your use case?
```python
import polars as pl

values = ["a", "b", "c"]
df = pl.DataFrame(pl.Series("x", values, dtype=pl.Enum(values))).to_pandas()

import duckdb as ddb

ddb.sql("SELECT * FROM df").show()
```
┌─────────────────────┐
│          x          │
│ enum('a', 'b', 'c') │
├─────────────────────┤
│ a                   │
│ b                   │
│ c                   │
└─────────────────────┘
```
